projects/com.oracle.truffle.llvm.types/src/com/oracle/truffle/llvm/types/LLVMFunctionDescriptor.java

```diff
@@ -45,6 +46,7 @@
 import com.oracle.truffle.api.nodes.RootNode;
 import com.oracle.truffle.llvm.types.memory.LLVMStack;
 
+@ValueType
 public final class LLVMFunctionDescriptor implements TruffleObject, Comparable<LLVMFunctionDescriptor> {
```

Comment:
How is this case different from `LLVMAddress`? Both are mainly runtime data structures. Also, `LLVMFunctionDescriptor` implements both `hashCode` and `equals` based on `functionId` field.